backend/backend.py

```diff
@@ -38,32 +39,32 @@ def start_video(self) -> None:
             video. """
 
     def on_base_canvas_updated(self, base_canvas: Canvas) -> None:
-        """ Catches any errors that occurs while the base canvas is being
+        """ Updates the base layer of the overlay whenever the canvas is updated,
+            which is shown on the camera.
+
+            Catches any errors that occurs while the base canvas is being
             updated. """
-        try:
+        with CameraException(self.client, self.backend_name):
```

Comment:
Rather than creating a new `CameraException` object every time we need error handling, we could store and instance of the class and only need to create it once in `__init__`. i.e.
```python
# in __init__
self.exceptionHandler = CameraException(client, self.backend_name)
# in functions
with self.exceptionHandler:
    # code
```
I haven't tested this, but I think it should work. Also would mean we don't need to store a reference to the `client`.

Reply:
Wait. That's actually brilliant